--- plugins/audio/SILK/SILK_SDK/Makefile.link	2011-06-14 10:38:53.000000000 +0200
+++ plugins/audio/SILK/SILK_SDK/Makefile	2011-07-20 20:09:48.000000000 +0200
@@ -35,7 +35,7 @@
 LDLIBS  += $(call ldlibs-from-libs,$(LIBS))
 
 COMPILE.c.cmdline   = $(CC) -c $(CFLAGS) -o $@ $<
-LINK.o.cmdline      = $(LINK.o) -lm $^ $(LDLIBS) -o $@$(EXESUFFIX) 
+LINK.o.cmdline      = $(LINK.o) $^ $(LDLIBS) -o $@$(EXESUFFIX) -lm
 ARCHIVE.cmdline     = $(AR) $(ARFLAGS) $@ $^ && $(RANLIB) $@
 
 %$(OBJSUFFIX):%.c
@@ -83,13 +83,13 @@
 $(TARGET): $(OBJS)
 	$(ARCHIVE.cmdline)
 
-encoder$(EXESUFFIX): $(ENCODER_OBJS)	
+encoder$(EXESUFFIX): $(ENCODER_OBJS) $(TARGET)
 	$(LINK.o.cmdline)
 
-decoder$(EXESUFFIX): $(DECODER_OBJS)	
+decoder$(EXESUFFIX): $(DECODER_OBJS) $(TARGET)
 	$(LINK.o.cmdline)
 
-signalcompare$(EXESUFFIX): $(SIGNALCMP_OBJS)	
+signalcompare$(EXESUFFIX): $(SIGNALCMP_OBJS) $(TARGET)
 	$(LINK.o.cmdline)
 
 clean:
--- plugins/plugin-inc.mak.in.link	2011-07-20 21:39:44.000000000 +0200
+++ plugins/plugin-inc.mak.in	2011-07-20 21:39:52.000000000 +0200
@@ -69,7 +69,7 @@
 OBJECTS	= $(addprefix $(OBJDIR)/,$(patsubst %.cxx,%.o,$(patsubst %.cpp,%.o,$(patsubst %.c,%.o,$(notdir $(SRCS))))))
 
 $(PLUGIN_PATH): $(SUBDIRS) $(OBJECTS)
-	$(Q_LD)$(CXX) $(LDFLAGS) -o $@ $(OBJECTS) $(LIBS)
+	$(Q_LD)$(CXX) $(LDFLAGS) -o $@ $(OBJECTS) $(LIBS) -lpthread -ldl
 
 ifneq ($(SUBDIRS),)
 .PHONY: $(SUBDIRS)
